src/app/settings/panel-settings/services/panel-input-modifier-state.service.ts

```diff
@@ -0,0 +1,43 @@
+import { Injectable } from '@angular/core'
+import { PanelInputModifier } from '../models/panel-input-modifier'
+import { BehaviorSubject } from 'rxjs'
+import { PanelHttpService } from './panel-http.service'
+import { PanelInputModifierEventObserver } from './panel-input-modifier-event-observer.service'
+
+@Injectable()
+export class PanelInputModifierStateService {
+  private panelInputModifiers: PanelInputModifier[] = []
+  private readonly panelInputModifiersSubject: BehaviorSubject<PanelInputModifier[]> = new BehaviorSubject(this.panelInputModifiers)
+
+  constructor(panelHttpService: PanelHttpService, panelInputModifierEventObserver: PanelInputModifierEventObserver) {
+    panelHttpService.getPanelInputModifiers().subscribe(panelInputModifiers => {
+      this.panelInputModifiers = panelInputModifiers
+      this.updateSubject()
+    })
+
+    panelInputModifierEventObserver.subscribeToPanelInputModifierCreated(event => this.addPanelInputModifier(event.panelInputModifier))
+    panelInputModifierEventObserver.subscribeToPanelInputModifierDeleted(event => this.removePanelInputModifier(event.panelInputModifierId))
+  }
+
+  private updateSubject(): void {
+    this.panelInputModifiersSubject.next(this.panelInputModifiers)
+  }
+
+  private addPanelInputModifier(panelInputModifier: PanelInputModifier): void {
+    this.panelInputModifiers.push(panelInputModifier)
+    this.updateSubject()
+  }
+
+  private removePanelInputModifier(panelInputModifierId: string): void {
+    const indexToDelete: number = this.panelInputModifiers.findIndex(modifier => modifier.id === panelInputModifierId)
+    if (indexToDelete === -1) {
+      return
+    }
+    this.panelInputModifiers.splice(indexToDelete, 1)
+    this.updateSubject()
+  }
```

Comment:
Just a suggestion :)

```suggestion
  private removePanelInputModifier(panelInputModifierId: string): void {
    this.panelInputModifiers = this.panelInputModifiers.filter(modifier => modifier.id !== panelInputModifierId)
    this.updateSubject()
  }
```